test/test_sum_product.py

```diff
@@ -143,8 +156,8 @@ def test_infinite_gradient(self):
         
         fgg = load_fgg('test/simplefgg.json')
         # make sum-product infinite
-        fgg.new_finite_factor('fac1', torch.tensor(1., requires_grad=True))
-        fgg.new_finite_factor('fac2', torch.tensor(1., requires_grad=True))
+        fgg.new_finite_factor('fac1', PatternedTensor(torch.tensor(1., requires_grad=True)))
```

Comment:
Is it no longer possible to create a factor out of a standard tensor? I think it would be good to.

Reply:
In the old code, the `weights` of a `FiniteFactor` could be any type: a list (of lists)* of floats or a tensor. The FGG code (such as `new_finite_factor`) would just stick whatever it gets into the `weights` attribute. Now I think the `weights` attribute is (or at least should) always be a `PatternedTensor`, so maybe I should change `FiniteFactor.__init__` to convert whatever it gets into a `PatternedTensor` always?

Reply:
I think that would be good. I thought we originally said we were going to make two different kinds of `Factor`s but that would probably result in more duplicated code.

Reply:
Ok, `FiniteFactor.weights` is now a property with a custom setter that converts to `PatternedTensor`

But now I should implement `PatternedTensor.__getitem__`